Entity-ngsiv2-schema.json

```diff
@@ -0,0 +1,185 @@
+{
+  "$schema": "http://json-schema.org/schema#",
+  "id": "https://fiware.github.io/dataModels/Entity-ngsiv2-schema.json",
+  "title": "NGSIv2 Entity",
+  "description": "NGSIv2 Entity",
+  "definitions": {
+    "EntityIdentifier": {
+      "type": "string",
+      "minLength": 1,
+      "maxLength": 256,
+      "pattern": "^[\\w\\-\\.\\{\\}\\$\\+\\*\\[\\]`|~^@!,:\\\\]+$"
+    },
+    "Name": {
+      "type": "string",
+      "pattern": "^(\\d|[a-zA-Z]|_)+$",
+      "minLength": 1,
+      "description": "NGSIv2 Name"
+    },
+    "Value": {
+      "type": [
+        "number",
+        "boolean",
+        "array",
+        "object",
+        "null",
+        "string"
+      ]
+    },
+    "TypeValue": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/Name"
+            },
+            {
+              "not": {
+                "const": "DateTime"
+              }
+            }
+          ]
+        },
+        "value": {
+          "$ref": "#/definitions/Value"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "DateTimeValue": {
+      "type": "string",
+      "format": "date-time"
+    },
+    "DateTimeTypeValue": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "const": "DateTime"
+        },
+        "value": {
+          "$ref": "#/definitions/DateTimeValue"
+        }
+      },
+      "required": [
+        "type",
+        "value"
+      ]
+    },
+    "GeoTypeValue": {
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "const": "geo:json"
+        },
+        "value": {
+          "$ref": "http://json-schema.org/geojson/geometry.json#"
```

Comment:
geo:point, geo:box, geo:line and geo:poligon are also valid types in attributes so they should be added to the JSON schema.

Reply:
I would refrain from that as they are not used in the FIWARE Data Models

Reply:
As far as a I understand this is a schema for NGSIv2 specification, idependently of its usage in FIWARE Data Models. Thus, as long as geo:point, geo:box, geo:line and geo:polygon are part of NGSIv2 specification, they should be inluded in the schema.

Reply:
well, we are in the context of the DataModels, this is the DataModels project ... 

NTC

Reply:
...but you are stating that this is an schema for NGSIv2. Thus, I'm the schema **is not a valid one for NGSIv2** is defined in the current way.

Reply:
If you want to use an incomplete schema then the following:

```
  "title": "NGSIv2 Entity. Normalized Format",
  "description": "NGSIv2 Entity",
```

should change to something like this:

```
  "title": "NGSIv2 Entity. Normalized Format. No geo:point, geo:box, geo:line, geo:polygon",
  "description": "NGSIv2 Entity. Normalized Format. No geo:point, geo:box, geo:line, geo:polygon",
```

and the name of the file should also change, for instance Entity-ngsiv2-partial-schema.json.

Reply:
Entity-ngsiv2-datamodels-schema.json

Reply:
Ok to the file name. However, description should be more cleaer. I have done a proposal in a separate comment.

NTC (with regards this comments thread)